InvenTree/order/migrations/0064_purchaseorderextraline_salesorderextraline.py

```diff
@@ -0,0 +1,103 @@
+# Generated by Django 3.2.12 on 2022-03-27 01:11
+
+import InvenTree.fields
+import django.core.validators
+from django.core import serializers
+from django.db import migrations, models
+import django.db.models.deletion
+import djmoney.models.fields
+import djmoney.models.validators
+
+
+def _convert_model(apps, line_item_ref, extra_line_ref, price_ref):
+    """Convert the OrderLineItem instances if applicable to new ExtraLine instances"""
+    OrderLineItem = apps.get_model('order', line_item_ref)
+    OrderExtraLine = apps.get_model('order', extra_line_ref)
+
+    items_to_change = OrderLineItem.objects.filter(part=None)
+
+    print(f'\nFound {items_to_change.count()} old {line_item_ref} instance(s)')
+    print(f'Starting to convert - currently at {OrderExtraLine.objects.all().count()} {extra_line_ref} / {OrderLineItem.objects.all().count()} {line_item_ref} instance(s)')
+    for lineItem in items_to_change:
+        newitem = OrderExtraLine(
+            order=lineItem.order,
+            notes=lineItem.notes,
+            price=getattr(lineItem, price_ref),
+            quantity=lineItem.quantity,
+            reference=lineItem.reference,
+        )
+        newitem.context = serializers.serialize('json', [lineItem, ])
```

Comment:
What is the purpose of this "context" field - what data / fields will be not available if we need to convert backwards?

Reply:
@schrodingersgat there are multiple values like target_date, fullfilment etc. I am also planning to use the context for more plugin related things.

Reply:
If these fields are still going to be used, I'd prefer to keep them in the new model. If not, I feel they should be deleted.

Can you give me an example of how the context would be used in the plugin system, differently to just keeping the fields against the model as "normal"?

Reply:
we do not need them in the new model - my thinking here is more that it does not add significant computational cost and there could be real information loss if a user updates without realising the model changed. There could be shipping target dates saved in them for example.

If a plugin wants to save additional data for a line the context could be great for that. Example: A line for shipping. Each time the plugin calculates the shipping rate, provider etc. it saves this information into the context of the line itself. I imagine this is more for documenting than for recalculating as it could give insights even after a plugin is removed.

Reply:
Fair, I can see that might be useful.